ee/modules/110-istio/hooks/hack_iop_reconciling.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2022 Flant JSC
+Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+*/
+
+package hooks
+
+import (
+	"time"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube/object_patch"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/deckhouse/deckhouse/ee/modules/110-istio/hooks/internal"
+	"github.com/deckhouse/deckhouse/ee/modules/110-istio/hooks/internal/crd"
+)
+
+type IstioOperatorCrdSnapshot struct {
+	Revision  string
+	NeedPunch bool
+}
+
+type IstioOperatorPodSnapshot struct {
+	Name           string
+	Revision       string
+	AllowedToPunch bool
+}
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Queue: internal.Queue("reconciling"),
+	Schedule: []go_hook.ScheduleConfig{
+		{Name: "cron", Crontab: "*/5 * * * *"},
+	},
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:              "istio_operators",
+			ApiVersion:        "install.istio.io/v1alpha1",
+			Kind:              "IstioOperator",
+			NamespaceSelector: internal.NsSelector(),
+			FilterFunc:        applyIopFilter,
+		},
+		{
+			Name:              "istio_operator_pods",
+			ApiVersion:        "v1",
+			Kind:              "Pod",
+			NamespaceSelector: internal.NsSelector(),
+			LabelSelector: &metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{
+						Key:      "revision",
+						Operator: "Exists",
+					},
+					{
+						Key:      "app",
+						Operator: "In",
+						Values:   []string{"operator"},
+					},
+				},
+			},
+			FilterFunc: applyIstioOperatorPodFilter,
+		},
+	},
+}, hackIopReconcilingHook)
+
+func applyIopFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var iop crd.IstioOperator
+	var result IstioOperatorCrdSnapshot
+	err := sdk.FromUnstructured(obj, &iop)
+	if err != nil {
+		return nil, err
+	}
+	result.Revision = iop.Spec.Revision
+	if iop.Status.Status == "ERROR" {
```

Comment:
1. Status.componentStatus.Pilot == "ERROR"
2. Add a regex to filter exact errors like `failed to update resource with server-side apply for obj EnvoyFilter`.

Reply:
fixed